### PR TITLE
Session Refactor

### PR DESCRIPTION
**Pull request by mitsuhiko**
Refactors HTTP session handling in Sentry.

This replaces #4943 

#nochanges

### CODE REVIEW

File: src/sentry/http.py [hunk 1]
```diff
@@ -108,25 +108,49 @@ def send(self, request, *args, **kwargs):
...
+        return response
+
+
+build_session = SafeSession
 
 
 def safe_urlopen(url, method=None, params=None, data=None, json=None,
                  headers=None, allow_redirects=False, timeout=30,
-                 verify_ssl=True, user_agent=None):
+                 verify_ssl=True, user_agent=None, session=None):
```
Comment (mattrobenolt) [inline src/sentry/http.py]: Is there a reason now to still allow this? Seems like you'd just use `SafeSession` directly if you want to reuse the Session object and `SafeSession` now contains all the logic that was repeated.

Comment (mitsuhiko) [inline src/sentry/http.py]: Yeah, i'm okay kicking this out.

**Comment by dcramer** [comment]
Curious what the need was here?

**Comment by mitsuhiko** [comment]
@dcramer for itunes connect we need to preload cookies from a cache into the session.

**Review by HazAT — APPROVED** [review]

**Pull request merged**